dask/async.py

```diff
@@ -422,8 +422,10 @@ def get_async(apply_async, num_workers, dsk, result, cache=None,
 
     if isinstance(result, list):
         result_flat = set(flatten(result))
-    else:
+    elif ishashable(result) and result in dsk:
         result_flat = set([result])
+    else:
+        return result
```

Comment:
This was needed to satisfy the pass through requirement

```
assert self.get(d, 'pass-through') == 'pass-through'
```
You are right about performance issues, I will push another commit with a possible alternative.